thirdeye/thirdeye-frontend/app/pods/components/yaml-editor/component.js

```diff
@@ -137,6 +281,116 @@ export default Component.extend({
     }
   }).drop(),
 
+  didRender(){
+    this._super(...arguments);
+
+    later(() => {
+      this._buildSliderButton();
+    });
+  },
+
+  // Helper function that builds the subchart region buttons
+  _buildSliderButton() {
+    const componentId = this.get('componentId');
+    const resizeButtons = d3.select(`.${componentId}`).selectAll('.resize');
+
+    resizeButtons.append('circle')
+      .attr('cx', 0)
+      .attr('cy', 30)
+      .attr('r', 10)
+      .attr('fill', '#0091CA');
+    resizeButtons.append('line')
+      .attr('class', 'anomaly-graph__slider-line')
+      .attr("x1", 0)
+      .attr("y1", 27)
+      .attr("x2", 0)
+      .attr("y2", 33);
+
+    resizeButtons.append('line')
+      .attr('class', 'anomaly-graph__slider-line')
+      .attr("x1", -5)
+      .attr("y1", 27)
+      .attr("x2", -5)
+      .attr("y2", 33);
+
+    resizeButtons.append('line')
+      .attr('class', 'anomaly-graph__slider-line')
+      .attr("x1", 5)
+      .attr("y1", 27)
+      .attr("x2", 5)
+      .attr("y2", 33);
+  },
+
+  _formatAnomaly(anomaly) {
+    return `${moment(anomaly.startTime).format(PREVIEW_DATE_FORMAT)}
+    to ${moment(anomaly.endTime).format(PREVIEW_DATE_FORMAT)}`;
+  },
+
+  _filterAnomalies(rows) {
+    return rows.filter(row => (row.startTime && row.endTime && !row.child));
+  },
+
+  _fetchTimeseries() {
+    const metricUrn = get(this, 'metricUrn');
+    const range = get(this, 'displayRange');
+    const granularity = '15_MINUTES';
+    const timezone = moment.tz.guess();
+
+    set(this, 'errorTimeseries', null);
+
+    const urlCurrent = `/rootcause/metric/timeseries?urn=${metricUrn}&start=${range[0]}&end=${range[1]}&offset=current&granularity=${granularity}&timezone=${timezone}`;
+    fetch(urlCurrent)
+      .then(checkStatus)
+      .then(res => {
+        set(this, 'timeseries', res);
```

Comment:
Anywhere you have a series of set or get to remember to use the set/getProperties method.